sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -201,7 +203,18 @@ def list_configuration_settings(
         """
 
     @distributed_trace
-    def list_configuration_settings(self, **kwargs) -> ItemPaged[ConfigurationSetting]:
+    def list_configuration_settings(self, *args, **kwargs) -> ItemPaged[ConfigurationSetting]:
```

Comment:
Yes, this is for back compat.

Reply:
Could you give more context/concrete sample why it is for back compat?

Reply:
We were supporting `key_filter` and `label_filter` as positional arguments: https://github.com/Azure/azure-sdk-for-python/pull/32021/files#diff-f1ef905e8030daf11ec8699b887dcad48b91d270810d48b03c1ded0e6c852b9a

Reply:
I am afraid adding *args cannot solve the back-compat problem. `key_filter` and `label_filter` are order sensitive positional args.

Reply:
Well. "only accept two positional args" to "accept arbitrary number of positional args" may be not breaking.

Reply:
I agree this change would allow customers to pass arbitrary number of positional args. In our implementation, we assume `*args` have two elements at most, and the order should be `key_filter` then `label_filter`, we'll ignore the rest values. So technically `*args` only works for the original surface.

Reply:
can someone pass `key_filter` and `label_filter` either as args or kwarg ? Are the samples updated to reflect our preferred method of usage so that over time the backwards compat folk are fewer?

Reply:
- Yes, either way works.
- I didn't update the sample. We have a sample showing how to use label_filter, and it's passing by kwarg:
https://github.com/Azure/azure-sdk-for-python/blob/8c2eee3d6a6cfb5cbb6e7496b85c4c9ef9572b3c/sdk/appconfiguration/azure-appconfiguration/samples/hello_world_advanced_sample.py#L53

Reply:
Seems like I added the comments into the wrong PR.

This code is still breaking:

```py
config_settings = client.list_configuration_settings("MyKey1", key_filter ="MyKey2").
```

We used to raise error. But now we silently use "MyKey1" and ignores "MyKey2"

Reply:
This is a good point! We should throw the same error in this case, will update.